### PR TITLE
3ds-SDL: Add support for C-Stick (added as two additional axes)

### PR DESCRIPTION
**Pull request by badda71**

### CODE REVIEW

**Comment by fincs** [comment]
The C-Stick is an analog stick at its core. Is there any reason why you added it as a hat instead of as two axes like the circle pad?

**Comment by badda71** [comment]
Initially, I wanted to add this with two axes. However, looking at the libctru documentation, I cannot find any function to read the C-Stick position (just as there is hidCircleRead() for the C-Pad). The only way to read the C-Stick is with the hidKeys...() functions. If you have a hint on how to read the C-Stick position, I would be happy to change the PR.

**Comment by fincs** [comment]
It's called `irrstCstickRead`: https://github.com/devkitPro/libctru/blob/master/libctru/include/3ds/services/irrst.h#L35-L39

(Note that hid automatically initializes/handles irrst for you)

**Comment by badda71** [comment]
Ok, changed the code to provide two additional axis values

File: src/joystick/n3ds/SDL_sysjoystick.c [hunk 1]
```diff
@@ -11,7 +11,9 @@
 #include "../../video/n3ds/SDL_n3dsvideo.h"
 
 int old_x = 0, old_y = 0;
+int old_cs_x = 0, old_cs_y = 0;
 u32 key_press, key_release = 0;
+extern void irrstCstickRead(circlePosition* pos);
```
Comment (fincs) [inline src/joystick/n3ds/SDL_sysjoystick.c]: It shouldn't be necessary to declare this function here. Assuming 3ds.h is included, it should already be present.

Comment (badda71) [reply src/joystick/n3ds/SDL_sysjoystick.c]: You're right - initially, I used the macro hidCstickRead which is defined in irrst.h a couple of lines below, and this throws a warning. However, irrstCstickRead is defined in the current standard 3ds-header already.

**Review by fincs — APPROVED** [review]

**Comment by fincs** [comment]
Updated 3ds-sdl package is now available through pacman, `(sudo) (dkp-)pacman -Syu` to get it as always :)

**Pull request merged**